assignments/week_4/week_4_exercises/project_4_4/README.md

```diff
@@ -0,0 +1,346 @@
+trying to catch up to a car that's driving away from us. If we start off with some error such that
+<img src="https://latex.codecogs.com/svg.latex?\inline&space;K_p&space;\,&space;e(t)" title="K_p \, e(t)" />
+is equal to the target car's velocity, then our car is going to be going at the same velocity as the target car, and
+the error isn't going to decrease.
+
+Intuitively though, if you were trying to catch up to a car and you saw that the error was still present after a while,
+you would begin to step harder on the throttle. The longer you see that the error is present, the harder you step on the
+throttle.
+
+One way to describe this behavior would be to say that we have a component of our control that is proportional
+to the *integral* of the error: As the error stays present for longer, we're going to increase our control effort.
+
+This is an *integral controller*, where the *control* is proportional to the __integral__ of the *error*.
+
+<p align="center">
+    <img
+    src="https://latex.codecogs.com/svg.latex?I_\text{out}&space;=&space;K_\text{i}&space;\,&space;\int_0^t&space;e(\tau)&space;\,d\tau"
+    title="I_\text{out} = K_\text{i} \, \int_0^t e(\tau) \,d\tau" />
+</p>
+
+It's easier to think of this in terms of **discrete** time, where time happens in small steps, as this is how a computer
+works. Instead of computing an integral, we compute a sum of the error and multiply it by the time step between each
+interval:
+
+<p align="center">
+    <img src="https://latex.codecogs.com/svg.latex?I_\text{out}&space;=&space;K_\text{i}&space;\,&space;\sum_{\tau=0}^{t}&space;e(\tau)&space;\,\Delta&space;t" title="I_\text{out} = K_\text{i} \, \sum_{\tau=0}^{t} e(\tau) \,\Delta t" />
+</p>
+
+or calculating recursively
+
+<p align="center">
+    <img src="https://latex.codecogs.com/svg.latex?\begin{align*}&space;I_\text{out}&space;=&&space;K_\text{i}&space;\cdot&space;\;&space;\textrm{accumulator}_t&space;\\&space;\textrm{accumulator}_t&space;=&&space;\;&space;\textrm{accumulator}_{t-1}&space;&plus;&space;e(t)&space;\;&space;\Delta&space;t&space;\end{align*}" title="\begin{align*} I_\text{out} =& K_\text{i} \cdot \; \textrm{accumulator}_t \\ \textrm{accumulator}_t =& \; \textrm{accumulator}_{t-1} + e(t) \; \Delta t \end{align*}" />
+</p>
+
+### Exercise: A moving target, and implementing an integral controller
+Let's try a moving target this time. Do this by using Ctrl-/ and commenting the `<param name="world_name" value="stationary" />` line
+and uncommenting the line below in the [project_4_4.launch](../launch/project_4_4.launch) file.
+
+It should look like this:
+```launch
+<launch>
+    <node pkg="buzzsim" type="buzzsim" name="buzzsim">
+        <param name="config_path" value="$(find buzzsim)/config/world.yml" />
+        <param name="world_name" value="stationary" />
+       <!-- <param name="world_name" value="moving" /> -->
```

Comment:
Wrong parameter is commented out

Reply:
How so? I want them to do `stationary` first

Reply:
This is supposed to be the launch code example for when you want it to be `moving`

Reply:
I want to run PID controller when the top turtle is stationary and then run it when the top turtle is moving.

Reply:
Yes... and this is the example launch code after you say they should be "commenting the `<param name="world_name" value="stationary" />` line and uncommenting the line below", but it's still has the "stationary" parameter uncommented and the "moving" one commented.

Reply:
I understand now. Fixed